client/home/lib/myteam/components/hometeamsendinvites/view.coffee

```diff
@@ -35,11 +42,22 @@ module.exports = class HomeTeamSendInvitesView extends React.Component
         onClick={@props.onInputChange.bind(null, rowIndex, 'canEdit', { target: { value: not checked}})}/>
     </div>
 
+  onInputChange: (rowIndex, fieldName, evt) ->
+    @state.filledInputCount[rowIndex] = if isEmailValid evt.target.value then 1 else 0
```

Comment:
assigning to state should be done via `setState` method, but it will trigger a new render, so to make it safer we should use the completion handler of `setState` method (a callback passed as 2nd argument):

```coffee
filledInputCount = _.assign {}, @state.filledInputCount
filledInputCount[rowIndex] = if isEmailValid evt.target.value then 1 else 0

@setState { filledInputCount }, => 
  @props.onInputChange rowIndex, 'email', evt
```

But this will trigger 2 render cycles, so i am not sure if this is the best way to approach this problem. I think moving `filledInputCount` to props, and managing it from the parent component would be a better approach.